### PR TITLE
Collect payment and Zipcode

### PR DESCRIPTION
**Pull request by dalthon**
Done!

### CODE REVIEW

File: lib/pagarme.rb [hunk 1]
```diff
@@ -24,6 +25,6 @@ class << self
   self.api_key      = ENV['PAGARME_API_KEY']
 
   def self.validate_fingerprint(id, fingerprint)
-    Digest::SHA1.hexdigest(id.to_s + "#" + api_key) == fingerprint
+    PagarMe::Postback.validate id, fingerprint
```
Comment (lucasgomide) [inline lib/pagarme.rb]: :+1: nice!

**Comment by dalthon** [comment]
@lucasgomide this solution you proposed was the first thing I thought when I realized that it could have timezone issues, but this library don't have active_support as a dependency.

I think that requiring it should be too much just to add a default value, my doubt is about providing a default value or not.

The same applies to implement a this method, manually just to set a default... Don't think it worth

What do you think?

**Comment by lucasgomide** [comment]
Hm... it's true "/

But sincerely I don't like this.
I agree that add `active_record` lib to add a default value is bad.. 

:cry:

**Comment by dalthon** [comment]
So, removed default, so people can handle it properly on their frameworks

**Comment by lucasgomide** [comment]
Okay, you'll add this on README ?

**Comment by dalthon** [comment]
Hm... I forgot to change that on readme, thanks!

In readme, since it's just an example I'll add something like:

```ruby
    Date.new(2020,4,1)
```

File: lib/pagarme/nested_model.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    end
+
+    class << self
+
+      def parent_resource_name
+        raise NotImplementedError
+      end
+
+      def find_by_id(parent_id, id)
+        raise RequestError.new('Invalid ID')        if id.nil?        ||        id == ''
```
Comment (lucasgomide) [inline lib/pagarme/nested_model.rb]: Check id String is `nil` or is `empty` is a implementation of the `blank?` look https://github.com/rails/rails/blob/42b0c942520e59399d70c2170253aa5275a42af1/activesupport/lib/active_support/core_ext/object/blank.rb#L17

So, you can apply one Monkey Patching in Object module.. creating a new method `blank?``

What do you think?

**Comment by dalthon** [comment]
Yeah, blank and present methods worth.

I'll refactor that. It's not just on this pull request, there are many other places were it applies.

Thanks!

**Comment by lucasgomide** [comment]
I know, for that i said this rsrs

Okay (:

**Comment by dalthon** [comment]
Done!

**Pull request merged**